File: tool/gen_icon.mjs
import { readFileSync, writeFileSync } from "fs";
import { dirname, join } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const root = join(__dirname, "..");
const svgPath = join(root, "assets", "logo.svg");
const outPath = join(root, "assets", "icon_app.png");
const BG = "#0098C4";
const SIZE = 1024;
const VB_W = 85.48, VB_H = 85.17;
const scale = Math.min(SIZE / VB_W, SIZE / VB_H);
const dx = (SIZE - VB_W * scale) / 2;
const dy = (SIZE - VB_H * scale) / 2;

const svg = readFileSync(svgPath, "utf8");
const inner = svg
  .replace(/^<\?xml[^>]*>\s*/, "")
  .replace(/<!DOCTYPE[^>]*>\s*/, "")
  .replace(/<svg[^>]*>/, "")
  .replace(/<\/svg>\s*$/, "");

const wrapped = `<svg xmlns="http://www.w3.org/2000/svg" width="${SIZE}" height="${SIZE}" viewBox="0 0 ${SIZE} ${SIZE}">
  <rect width="100%" height="100%" fill="${BG}"/>
  <g transform="translate(${dx}, ${dy}) scale(${scale})">
    ${inner}
  </g>
</svg>`;

const sharp = (await import("sharp")).default;
await sharp(Buffer.from(wrapped))
  .png()
  .toFile(outPath);

console.log(`Generated ${outPath} (${SIZE}x${SIZE})`);

File: tool/gen_icon.py
"""Generate icon_app.png from logo.svg (1024x1024, #0098C4 background)."""
import cairosvg
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
SVG = ROOT / "assets" / "logo.svg"
OUT = ROOT / "assets" / "icon_app.png"
BG = "#0098C4"
SIZE = 1024

if not SVG.exists():
    raise SystemExit(f"ERROR: {SVG} not found")

cairosvg.svg2png(
    url=str(SVG),
    write_to=str(OUT),
    output_width=SIZE,
    output_height=SIZE,
    background_color=BG,
)
print(f"Generated {OUT} ({SIZE}x{SIZE})")
